pytorch_lightning/trainer/trainer.py

```diff
@@ -466,7 +473,7 @@ def __init__(
         if show_progress_bar is not None:
             self.show_progress_bar = show_progress_bar
 
-        self.progress_bar_refresh_rate = progress_bar_refresh_rate
+        self.progress_bar_refresh_rate = self.configure_progress_bar_distrib(progress_bar_refresh_rate)
```

Comment:
this shall be general with `rank_aero_only` @jeremyjordan ?

Reply:
Currently, we set `rank_zero_only.rank` at the start of the train process, but this check is performed during initialization, so we would need to refactor that.  It might be a little confusing, because Horovod is in a multi-rank context even during initialization, while DDP is not.

Reply:
If it's not possible, as you say, I would disable the progressbar callback with 
`self.progress_bar_callback.disable()` as it is currently done in case of ddp. 
Because currently you set the refresh rate in the trainer, but this has no effect after the progress_bar_callback was initialized (it has the refresh rate internally). 
We should probably make the refresh rate a protected attribute anyway.

Reply:
Maybe @Borda or @jeremyjordan  has better suggestion, I'm just commenting here because I made the progress bar callback. Maybe wait for what they have to say ;)

Reply:
That's a good suggestion, @awaelchli.  I'll take another look and see if I can make that work.